packages/commonwealth/client/scripts/controllers/server/sessions.ts

```diff
@@ -32,298 +31,226 @@ export class SessionKeyError extends Error {
 export async function signSessionWithAccount<T extends { address: string }>(
   wallet: IWebWallet<T>,
   account: Account,
-  timestamp: number,
-) {
-  // Try to infer Chain ID from the currently active chain.
-  // `chainBaseToCanvasChainId` will replace idOrPrefix with the
-  // appropriate chainID for non-eth, non-cosmos chains.
-  //
-  // However, also handle the case where app.chain is empty.
-  const idOrPrefix =
-    wallet.chain === ChainBase.CosmosSDK
-      ? app.chain?.meta.bech32Prefix || 'cosmos'
-      : app.chain?.meta.node?.ethChainId || 1;
-  const canvasChainId = chainBaseToCanvasChainId(wallet.chain, idOrPrefix);
-  const sessionPublicAddress = await app.sessions.getOrCreateAddress(
-    wallet.chain,
-    canvasChainId,
-    account.address,
-  );
-
-  const sessionPayload = createCanvasSessionPayload(
-    wallet.chain,
-    canvasChainId,
-    wallet.chain === ChainBase.Substrate
-      ? addressSwapper({
-          address: account.address,
-          currentPrefix: 42,
-        })
-      : account.address,
-    sessionPublicAddress,
-    timestamp,
-    account.validationBlockInfo
-      ? JSON.parse(account.validationBlockInfo).hash
-      : null,
-  );
-
-  const signature = await wallet.signCanvasMessage(account, sessionPayload);
-  return {
-    signature,
-    chainId: canvasChainId,
-    sessionPayload,
-  };
-}
-
-// for eth and cosmos only, assumes chainbase is either Ethereum or CosmosSDK
-export async function signSessionWithMagic(
-  walletChain = ChainBase.Ethereum,
-  signer,
-  signerAddress,
-  timestamp: number,
-  blockhash = '',
 ) {
-  const idOrPrefix =
-    walletChain === ChainBase.CosmosSDK
-      ? app.chain?.meta.bech32Prefix || 'cosmos'
-      : app.chain?.meta.node?.ethChainId || 1;
-  const canvasChainId = chainBaseToCanvasChainId(walletChain, idOrPrefix);
-  const sessionPublicAddress = await app.sessions.getOrCreateAddress(
-    walletChain,
-    canvasChainId,
-    signerAddress,
-  );
-
-  const sessionPayload = createCanvasSessionPayload(
-    walletChain,
-    canvasChainId,
-    walletChain === ChainBase.Substrate
-      ? addressSwapper({
-          address: signerAddress,
-          currentPrefix: 42,
-        })
-      : signerAddress,
-    sessionPublicAddress,
-    timestamp,
-    blockhash,
-  );
-
-  // skip wallet.signCanvasMessage(), do the logic here instead
-  if (walletChain === ChainBase.CosmosSDK) {
-    const canvas = await import('@canvas-js/interfaces');
-    const { msgs, fee } = await getADR036SignableSession(
-      Buffer.from(canvas.serializeSessionPayload(sessionPayload)),
-      signerAddress,
+  const session = await getSessionFromWallet(wallet);
```

Comment:
When the session is expired manually from local storage, we are still able to create a thread (or any auth requiring action), before page reload. Seems like we are loading the session value once on app mount? If yes, shouldn't we access the latest value of session before performing the auth actions?

Reply:
Our signers are caching sessions in localStorage, yeah: https://github.com/canvasxyz/canvas/blob/81ea4a45efa3a8d34759c2d92bb0066d1578a934/packages/signatures/src/AbstractSessionSigner.ts#L58

Strictly speaking, localStorage isn't supposed to be used as an interface to session storage, it's just the most convenient way to clear session data (with a refresh), since we haven't implemented a "clear sessions" or "clear cache" menu option.

Reply:
> Strictly speaking, localStorage isn't supposed to be used as an interface to session storage, it's just the most convenient way to clear session data

I agree, but just to confirm if this would cause any security issues because of using expired cache value?

Reply:
Ah, there shouldn't be any issues here. Expiration happens when sessions are loaded from the session cache in JS (not in localStorage) which is unaffected.